xlators/protocol/server/src/server.c

```diff
@@ -1913,6 +1932,9 @@ struct volume_options server_options[] = {
      .default_value = "off",
      .description = "strict-auth-accept reject connection with out"
                     "a valid username and password."},
+    {.key = {"volspec-directory", "volfile-path"},
+     .type = GF_OPTION_TYPE_PATH,
+     .default_value = GLUSTERD_DEFAULT_WORKDIR "/vols"},
```

Comment:
This patch publishes internal server information without any synchronization with potential concurrent changes made by glusterd, so this data can be completely incorrect unless you are sure no one else is modifying the volume concurrently. Since we cannot guarantee this from the brick side, I think it would be better to set the default value to NULL to indicate that this feature is disabled (and no spec file will be served). Only if explicitly configured it should serve spec files.

Reply:
Agree. We can have different path other than existing Glusterd/vols path.

Reply:
Even with a different path it seems like an unnecessary exposure if the feature won't be used. I would define it as disabled by default and only enable it if really required.